src/components/spreadsheet/spreadsheet.ts

```diff
@@ -226,13 +227,19 @@ export class Spreadsheet extends Component<SpreadsheetProps, SpreadsheetChildEnv
     });
 
     useExternalListener(window as any, "resize", () => this.render(true));
-    useExternalListener(window, "beforeunload", this.unbindModelEvents.bind(this));
+    useExternalListener(window, "beforeunload", () => this.unbindModelEvents(this.model));
 
-    this.bindModelEvents();
+    this.bindModelEvents(this.props.model);
     onMounted(() => {
       this.checkViewportSize();
     });
-    onWillUnmount(() => this.unbindModelEvents());
+    onWillUpdateProps((nextProps) => {
+      if (nextProps.model !== this.props.model) {
+        this.unbindModelEvents(this.props.model);
+        this.bindModelEvents(nextProps.model);
+      }
```

Comment:
it will become an issue once we introduce the history manager which will regenerate new models each time we jump in time. I realized that I missed some other occurences in the meantime :/